### PR TITLE
edm4hep: Add v0.99 release and deprecate older versions

### PR DESCRIPTION
**Pull request by tmadlener**

### CODE REVIEW

**Review by jmcarcell — APPROVED** [review]

**Comment by wdconinc** [comment]
I don't agree with marking older versions as deprecated for no good reason. Preference is already given to newer versions when users decide to upgrade.

Please consider keeping the previous practice (as in use in other packages) of only marking versions as deprecated for reasons of security or functional issues, and to reduce technical debt a long time after release.

**Comment by tmadlener** [comment]
In this case I did deliberately mark these deprecated, because between `0.10.5` and `0.99` we introduced quite a few breaking changes and we don't really have the intention to ever make that transition smooth. This includes the fact, that files written with `0.10.5` will not be readable with `0.99`. Hence, we really want people to move off of those versions as quickly as possible.

On the other hand we consider `0.99` as a pre-release and will keep compatibility with that beyond `1.0`.

**Comment by wdconinc** [comment]
> files written with 0.10.5 will not be readable with 0.99

In that case I'm even more strongly against this change. You are breaking the implicit semver promise and leaving people with files they can now not read with the newer version nor install the older version.

File: var/spack/repos/builtin/packages/edm4hep/package.py [hunk 1]
```diff
@@ -21,16 +21,37 @@ class Edm4hep(CMakePackage):
     license("Apache-2.0")
 
     version("main", branch="main")
-    version("0.10.5", sha256="003c8e0c8e1d1844592d43d41384f4320586fbfa51d4d728ae0870b9c4f78d81")
+    version("0.99", sha256="3636e8c14474237029bf1a8be11c53b57ad3ed438fd70a7e9b87c5d08f1f2ea6")
+    version(
+        "0.10.5",
+        sha256="003c8e0c8e1d1844592d43d41384f4320586fbfa51d4d728ae0870b9c4f78d81",
+        deprecated=True,
+    )
```
Comment (tmadlener) [inline var/spack/repos/builtin/packages/edm4hep/package.py]:
```suggestion
    version("0.10.5", sha256="003c8e0c8e1d1844592d43d41384f4320586fbfa51d4d728ae0870b9c4f78d81")
```

**Comment by tmadlener** [comment]
I have "undeprecated" 0.10.5 here so that people can still build it, at least for some time. At some point in the (hopefully not too distant) future, all Key4hep packages will depend on EDM4hep v1.0 that actually respects SemVer for breaking changes instead of using the `0.x` grey area of things.

Just for completeness (although it doesn't change too much in the end): The readability between `0.10.5` and `0.99` depends on which datatypes are actually stored in the file. Some of them will work just fine, but for others there are breaking changes.

**Review by wdconinc — APPROVED** [review]
Thanks!

**Pull request merged**